glhe/properties/fluid_properties.py

```diff
@@ -1,237 +1,92 @@
-from CoolProp.CoolProp import PropsSI
-from numpy import arange
+from __future__ import annotations
 
-from glhe.properties.fluid_property_types import FluidPropertyType
-from glhe.properties.fluid_types import FluidType
-from glhe.utilities.functions import c_to_k, k_to_c, Interpolator1D
+from scp.ethyl_alcohol import EthylAlcohol
+from scp.ethylene_glycol import EthyleneGlycol
+from scp.methyl_alcohol import MethylAlcohol
+from scp.propylene_glycol import PropyleneGlycol
+from scp.water import Water
 
 
 class Fluid:
     def __init__(self, inputs: dict):
         self.type = inputs['fluid-type'].upper()
         if self.type == "WATER":
-            self.fluid_enum = FluidType.WATER
             concentration = 0
+            self.fluid = Water()
         elif self.type == "EA":
-            self.fluid_enum = FluidType.ETHYL_ALCOHOL
             concentration = inputs['concentration'] / 100.0
+            self.fluid = EthylAlcohol(concentration)
         elif self.type == "EG":
-            self.fluid_enum = FluidType.ETHYLENE_GLYCOL
             concentration = inputs['concentration'] / 100.0
+            self.fluid = EthyleneGlycol(concentration)
+        elif self.type == "MA":
+            concentration = inputs['concentration'] / 100.0
+            self.fluid = MethylAlcohol(concentration)
         elif self.type == "PG":
-            self.fluid_enum = FluidType.PROPYLENE_GLYCOL
             concentration = inputs['concentration'] / 100.0
+            self.fluid = PropyleneGlycol(concentration)
         else:
             raise ValueError("Fluid '{}' fluid is not valid.".format(self.type))
 
-        self.fluid_str = self.get_fluid_str(self.fluid_enum, concentration)
-        self.min_temp = k_to_c(self.calc_min_temp())
-        self.max_temp = k_to_c(self.calc_max_temp())
-        self.pressure = 101325
-
-        temps = arange(self.min_temp, self.max_temp, 0.5)
-
-        cp_vals = [self.calc_specific_heat(x) for x in temps]
-        k_vals = [self.calc_conductivity(x) for x in temps]
-        mu_vals = [self.calc_viscosity(x) for x in temps]
-        pr_vals = [self.calc_prandtl(x) for x in temps]
-        rho_vals = [self.calc_density(x) for x in temps]
-        rho_cp_vals = [self.calc_vol_heat_capacity(x) for x in temps]
-
-        self.cp_interp = Interpolator1D(temps, cp_vals)
-        self.k_interp = Interpolator1D(temps, k_vals)
-        self.mu_interp = Interpolator1D(temps, mu_vals)
-        self.pr_interp = Interpolator1D(temps, pr_vals)
-        self.rho_interp = Interpolator1D(temps, rho_vals)
-        self.rho_cp_interp = Interpolator1D(temps, rho_cp_vals)
```

Comment:
@Myoldmopar a whole bunch of these interpolation usages were here because of this. This eliminates it and just calls the fluid class to make calculations each time. Maybe there's some behind-the-scenes caching happening, but I'm not smart enough to know whether it's happening.